lib/login/google_Sign_In.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+class UserAuth {
+  UserAuth({
+    @required this.googleSignIn,
+    @required this.firebaseAuth,
+  });
+
+  GoogleSignIn googleSignIn;
+  FirebaseAuth firebaseAuth;
+
+  Future<FirebaseUser> signInWithGoogle() async {
+    FirebaseUser user;
+    final bool isSignedIn = await googleSignIn.isSignedIn();
+    if (isSignedIn) {
+      user = await firebaseAuth.currentUser();
+    } else {
+      final GoogleSignInAccount googleUser = await googleSignIn.signIn();
+      final GoogleSignInAuthentication googleAuth =
+          await googleUser.authentication;
+
+      final AuthCredential credential = GoogleAuthProvider.getCredential(
+          idToken: googleAuth.idToken, accessToken: googleAuth.accessToken);
+
+      user = (await firebaseAuth.signInWithCredential(credential)).user;
+    }
+
+    return user;
+  }
+
+  Future<String> onGoogleSignIn(BuildContext context) async {
```

Comment:
There are issues with code readability due to the function names here. I think it would be better if you renamed this method to signInWithGoogle()

Reply:
I already have a method named signInWithGoogle that authenticates that the email is inside the Firebase database. I have it named "onGoogleSignIn" because that is the method that initiates the process of the user logging in with google. Is there another name suggestion?